modules/fileout_netcdf/FONcArrayStructure.cc

```diff
@@ -0,0 +1,220 @@
+// Suite 200, Champaign, IL 61820
+
+#include <BESInternalError.h>
+#include <BESDebug.h>
+
+#include "FONcArrayStructure.h"
+#include "FONcUtils.h"
+#include "FONcAttributes.h"
+
+/** @brief Constructor for FONcArrayStructure that takes a DAP Array Structure
+ *
+ * This constructor takes a DAP BaseType and makes sure that it is a DAP Array
+ * Structure instance. If not, it throws an exception
+ *
+ * @param b A DAP BaseType that should be an array of Structure
+ * @throws BESInternalError if the BaseType is not a an array
+ */
+FONcArrayStructure::FONcArrayStructure(BaseType *b) :
+    FONcBaseType()
+{
+    _as = dynamic_cast<Array *>(b);
+    if (!_as) {
+        auto s = (string) "File out netcdf, array of structure was passed a " + "variable that is not an array ";
+        throw BESInternalError(s, __FILE__, __LINE__);
+    }
+    if (_as->var()->type() != dods_structure_c) {
+        string s = (string) "File out netcdf, array of structure was passed a " + "variable that is not a structure ";
+        throw BESInternalError(s, __FILE__, __LINE__);
+    }
+}
+
+/** @brief Destructor that cleans up the array of structure
+ *
+ * Delete each of the FONcBaseType instances that is a part of this
+ * structure.
+ */
+FONcArrayStructure::~FONcArrayStructure()
+{
+
+    // Adapt the code from FONcStructure.cc
+    bool done = false;
+    while (!done) {
+        // _vars are vector<FONcArrayStructureField *>
+        auto i = _vars.begin();
+        auto e = _vars.end();
+        if (i == e) {
+            done = true;
+        }
+        else {
+            FONcArrayStructureField *b = (*i);
+            delete b;
+            _vars.erase(i);
+        }
+    }
+
+}
+
+/** @brief Creates the FONc objects for each variable of the DAP array of structure
+ *
+ * For each of the variables of the DAP Structure we convert to a
+ * similar FONc object. We still follow the current way to map structure to netCDF-3.
+ * We must flatten out the structures. To do this, we embed the name of the
+ * structure as part of the name of the children variables. For example,
+ * if the structure, called s1, contains an array called a1 and an int
+ * called i1, then two variables are created in the netcdf file called
+ * s1.a1 and s1.i1.
+ *
+ * @note This method only converts the variables that are to be sent. This keeps
+ * the convert() and write() methods below from operating on DAP variables
+ * that should not be sent.
+ *
+ * @param embed The parent names of this structure.
+ * @throws BESInternalError if there is a problem converting this
+ * structure
+ */
+void FONcArrayStructure::convert(vector<string> embed, bool _dap4, bool is_dap4_group){
+    set_is_dap4(_dap4);
+    FONcBaseType::convert(embed,_dap4,is_dap4_group);
+    embed.push_back(name());
+
+    auto as_v = dynamic_cast<Structure *>(_as->var());
+    if (!as_v) {
+        string s = (string) "File out netcdf, write an array of structure was passed a " + "variable that is not a structure";
```

Comment:
You can drop the '+' since it's all a constant.

Reply:
Yes. change it.